climada/engine/impact.py

```diff
@@ -1372,7 +1372,147 @@ def _selected_exposures_idx(self, coord_exp):
             LOGGER.warning("No exposure coordinates match the selection.")
         return sel_exp
 
+    @classmethod
+    def concat(cls, imp_list,concat_type='time_window'):
+        """Concatenate impact objects of the same hazard type, frequency, 
+        exposure, and coordinate system.
 
+        Parameters
+        ----------
+        imp_list : list of climada.engine.Impact objects
+                list of Impact object to concatenate
+        concat_type: string
+                string specifying the type of concatenation to be done.
+                    'time_window': concatenate impacts for different time windows / time slices within a year (e.g. May to June, July to August,...)
+                    'exposure': concatenate impacts with separate exposures
+        Returns
+        --------
+        impact: climada.engine.Impact
+               New impact object which is a concatenation of all impacts
+        """
+       
+        if len(imp_list) == 0:
+           return cls()
+       
+        if concat_type not in ['time_window','exposure']:
+           raise ValueError(f"Concatenation option {concat_type} not available.")
+       
+        #initialize new impact object
+        impact=cls()
+       
+        # check impacts consistency and fill impact object
+       
+        ### the following checks are common for both concat options
+       
+        # check hazard type
+        haz_types = {imp.tag['haz'].haz_type for imp in imp_list if imp.tag['haz'].haz_type != ''}
+        if len(haz_types) > 1:
+           raise ValueError(f"The impacts are based on hazards of different types: {haz_types}. "
+                            "The impacts are incompatible and cannot be concatenated.")      
+       
+        # check unit
+        units = {imp.unit for imp in imp_list if imp.unit != ''}
+        if len(units) > 1:
+           raise ValueError(f"The given impacts have different units: {units}. "
+                            "The impacts are incompatible and cannot be concatenated.")
+        if len(units) == 0:
+           units = {''}
+        impact.unit = units.pop()
+
+        #check coordinate reference system
+        crss={imp.crs for imp in imp_list if imp.crs != ''}
```

Comment:
```suggestion
        crs={imp.crs for imp in imp_list if imp.crs != ''}
```

Reply:
Why `crss` ?

Reply:
I used `crss` as a grammatically awful expression of the plural of `crs`, because it can be a dictionary with more than one crs, similarly to the `units` dictionary.

Reply:
I then suggest to drop the plural form. This just creates confusion.